jzp-code/kukdo/P2938_0606.java

```diff
@@ -0,0 +1,18 @@
+class Solution {
+    public long minimumSteps(String s) {
+        char[] num = s.toCharArray();
+        long step = 0;
+        long count = 0;
+        String subStr;
+        int index = num.length;
+        for (int i = num.length - 1; i >= 0; i--) {
+            if (num[i] == '1'){
+                subStr = s.substring(i+1, index);
+                index = i;
+                count += subStr.length();

```

Comment:
it could be 
```
count += index - i;
```
right? there is no need to slice the string (in Java, it has a time complexity of `O(n)`) if you just want the length of the slice.

Reply:
yep, indeed there is no need to copy the substring, just subtract the indexes like you mentioned. The difficulty with switching languages is being unskilled with many features and libraries, and the best way to do that is indeed to practice. Thanks for reviewing!